obspy/taup/seismic_phase.py

```diff
@@ -1655,6 +1655,32 @@ def calc_incident_angle(self, ray_param):
     def get_earliest_arrival(cls, rel_phases, degrees):
         raise NotImplementedError("baaa")
 
+    def _shadow_zone_splits(self):
+        """
+        A list of slices that split on any shadow zones.
+        """
+        if self.head_or_diffract_seq:
+            # don't split any diffracted waves
+            return [slice(0, None)]
+        else:
+            # shadow zones are where ray parameter repeats
+            return _repeated_value_splits(self.ray_param)
+
+
+def _repeated_value_splits(x):
+    """
+    A list of slices that split a 1D numpy array where values repeat.
+    """
+    split_indices = []
+    idxs = np.where(x[1:] - x[:-1] == 0)[0] + 1
```

Comment:
looks like this might be equivalent to

```python
idxs = np.diff(x) == 0
```

which might be a bit easier to read. but it's just cosmetics, so doesn't bother me too much.

Reply:
ah nevermind, you'd need to wrap around `np.argwhere` or something so lets just leave as is :smile:

Reply:
ah but `x[1:] - x[:-1]` is equivalent to `np.diff(x)`, let me quickly do that and merge then